include/zephyr/types.h

```diff
@@ -29,6 +29,36 @@ typedef union {
 	void            (*thepfunc)(void);
 } z_max_align_t;
 
+/*
+ * Offset type
+ *
+ * k_off_t, much like the POSIX off_t, is a signed integer type used to represent file sizes and
+ * offsets.
+ *
+ * k_off_t effectively limits the size of the file that can be handled by the system, it is
+ * therefore not tied to the word-size of the architecture.
+ *
+ * In order to overcome historical limitations of the off_t type, Zephyr always deefines
+ * k_off_t as a 64-bit signed integer.
+ */
+typedef int64_t k_off_t;
+
+/*
+ * Signed size type
+ *
+ * k_ssize_t, much like the POSIX ssize_t, is a signed integer type that effectively limits the
+ * size of I/O operations on files of returning negative error codes while still supporting
+ * negative values.
+ *
+ * k_ssize_t is tied to the word-size of the architecture. For compatibility purposes, k_ssize_t
+ * is defined as a 'long' on 64-bit architectures, and as an 'int' on 32-bit architectures.
+ */
+#ifdef CONFIG_64BIT
```

Comment:
It seems like you explicitly need this to match the type of `size_t`, except signed? What newlib/picolibc do is a clever hack which works on both clang and gcc:
```
#ifdef __SIZE_TYPE__
/* If __SIZE_TYPE__ is defined (gcc) we define ssize_t based on size_t.
   We simply change "unsigned" to "signed" for this single definition
   to make sure ssize_t and size_t only differ by their signedness. */
#define unsigned signed
typedef __SIZE_TYPE__ _ssize_t;
#undef unsigned
#else
...
```
Stick your existing definitions inside the `#else` clause.

Reply:
Should be fixed now